tests/mocks.py

```diff
@@ -748,6 +746,137 @@ def to_response(i):
+            "phabricator:auditors",
+            "bugzilla.bug-id",
+            "comment",
+        ]
+
+        if transactions is None or (
+            not isinstance(transactions, list) and not isinstance(transactions, dict)
+        ):
+            error_info = 'Parameter "transactions" is not a list of transactions.'
+            raise PhabricatorAPIException(
+                error_info, error_code="ERR-CONDUIT-CORE", error_info=error_info
+            )
+
+        # Normalize transactions into a unified format (they can either be a
+        # list or a dict). Internally phabricator treats lists as ordered
+        # dictionaries (php arrays) where the keys are integers.
+        if isinstance(transactions, list):
+            transactions = list(enumerate(transactions))
+        elif isinstance(transactions, dict):
+            transactions = list((k, v) for k, v, in transactions.items())
+
+        # Validate each transaction.
+        for key, t in transactions:
+            if not isinstance(t, dict):
+                error_info = f'Parameter "transactions" must contain a list of transaction descriptions, but item with key "{key}" is not a dictionary.'  # noqa
+                raise PhabricatorAPIException(
+                    error_info, error_code="ERR-CONDUIT-CORE", error_info=error_info
+                )
+
+            if "type" not in t:
+                error_info = f'Parameter "transactions" must contain a list of transaction descriptions, but item with key "{key}" is missing a "type" field. Each transaction must have a type field.'  # noqa
+                raise PhabricatorAPIException(
+                    error_info, error_code="ERR-CONDUIT-CORE", error_info=error_info
+                )
+
+            if t["type"] not in TRANSACTION_TYPES:
+                given_type = t["type"]
+                valid_types = " ,".join(TRANSACTION_TYPES)
+                error_info = f'Transaction with key "{key}" has invalid type "{given_type}". This type is not recognized. Valid types are: {valid_types}.'  # noqa
+                raise PhabricatorAPIException(
+                    error_info, error_code="ERR-CONDUIT-CORE", error_info=error_info
+                )
+
+        if objectIdentifier is None:
+            # A revision is being created, it must have the "title" and "update"
+            # transactions present.
+            transaction_types = [t[1]["type"] for t in transactions]
+            if "title" not in transaction_types or "update" not in transaction_types:
+                error_info = "Validation errors:"
+                if "title" not in transaction_types:
+                    error_info = error_info + "\n  - Revisions must have a title."
+                if "update" not in transaction_types:
+                    error_info = (
+                        error_info
+                        + "\n  - "
+                        + "You must specify an initial diff when creating a revision."
+                    )
```

Comment:
elsewhere you let the line run long and used `# noqa`.

the approach to long lines should be consistent - my vote is to concatenate instead of using `# noqa`.